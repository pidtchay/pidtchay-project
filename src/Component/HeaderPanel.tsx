import { Row, Col, Avatar, Button } from 'antd';
import { loadLang } from 'i18n/i18n';
import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
    MenuUnfoldOutlined,
    LoginOutlined,
    MenuFoldOutlined,
    UserOutlined,
    HomeOutlined
} from '@ant-design/icons';
import { ESupportedLanguages } from 'Constants/Common';
import { MenuRoute } from 'Constants/Routes';
import { RootState } from 'Store';
import { loadLiterals } from 'Store/literals/actions';
import { cleanSession } from 'Store/system/actions';
import style from 'Style/HeaderPanel.less';
import { getI18nValue } from 'Utils/common';
import history from 'Utils/history';

interface IHeaderPanel {
    isSideMenuToogle: boolean;
    onToggleTrueFalse: () => void;
}

// TODO Switch language button
export const HeaderPanel: React.FC<IHeaderPanel> = ({
    isSideMenuToogle,
    onToggleTrueFalse
}) => {
    const [language, setLanguage] = useState(ESupportedLanguages.RU);
    const [languageUsed, setLanguageUsed] = useState(
        ESupportedLanguages.EN.toUpperCase()
    );
    const auth = useSelector((state: RootState) => state.system.authenticated);
    const literals = useSelector((state: RootState) => state.literals);
    const dispatch = useDispatch();

    const onGoHome = () => history.push(MenuRoute.HOME);

    const changeLanguage = () => {
        setLanguage(
            language === ESupportedLanguages.EN
                ? ESupportedLanguages.RU
                : ESupportedLanguages.EN
        );
        setLanguageUsed(language.toUpperCase());
        // tslint:disable-next-line: no-floating-promises
        loadLang(language).then((lang) => dispatch(loadLiterals(lang)));
    };
    return (
        <div>
            <Row justify="space-between">
                <Col span={4}>
                    <Row justify="space-between">
                        <Col>
                            {React.createElement(
                                isSideMenuToogle
                                    ? MenuUnfoldOutlined
                                    : MenuFoldOutlined,
                                {
                                    className: 'trigger',
                                    onClick: onToggleTrueFalse
                                }
                            )}
                        </Col>
                        <Col>
                            <Button
                                type="link"
                                icon={<HomeOutlined />}
                                onClick={onGoHome}
                            >
                                PIDTCHAY PROJECT.
                            </Button>
                        </Col>
                    </Row>
                </Col>
                <Col span={5}>
                    <div className={style.header_layout}>
                        <Button
                            type="primary"
                            onClick={changeLanguage}
                            className={style.header_layout_i18n}
                        >
                            {languageUsed}
                        </Button>
                        {auth && (
                            <Avatar
                                size="large"
                                icon={<UserOutlined />}
                                className={style.header_layout_avatar}
                            />
                        )}
                        {auth && (
                            <Button
                                type="primary"
                                shape="round"
                                icon={<LoginOutlined />}
                                size="large"
                                className={style.header_layout_button}
                                onClick={() => dispatch(cleanSession())}
                            >
                                {getI18nValue(literals, 'ACTIONS.logout')}
                            </Button>
                        )}
                    </div>
                </Col>
            </Row>
        </div>
    );
};
