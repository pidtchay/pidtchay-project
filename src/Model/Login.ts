/**
 * Login data model.
 * @property [string] nickName - user nickName
 * @property [string] email - user email
 * @property [string] password - user password
 */
export interface ILoginData {
    nickName: string;
    email: string;
    password: string;
}