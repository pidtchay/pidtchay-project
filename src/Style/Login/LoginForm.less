.loginForm {
    max-width: 500px;
    width: 100%;
    margin: 0 auto;
    display: flex;
    flex-direction: column;
}
