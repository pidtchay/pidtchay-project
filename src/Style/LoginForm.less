.loginForm {
    max-width: 500px;
    width: 100%;
    margin: 0 auto;
    display: flex;
    flex-direction: column;

    .alert {
        background-color: rgb(70, 145, 156);
        border: 1px solid rgb(70, 173, 156);
        color: white;
        border-radius: 10px;
        margin: 10px 0px;
        text-align: center;
        padding: 15px;
    }

    .submitButton {
        padding: 10px 15px;
        background-color: rgb(70, 153, 179);
        color: white;
        border: 1px solid rgb(70, 153, 179);
        border-radius: 6px;
        background-color: 250ms;
    }

    .submitButton:hover {
        cursor: pointer;
        background-color: white;
        color: rgb(70, 153, 179);
    }
}