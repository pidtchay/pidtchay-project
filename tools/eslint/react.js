module.exports = {
    'react/no-did-update-set-state': 0,
    'react-hooks/exhaustive-deps': 0,
    'react/state-in-constructor': [2, 'never'],
    'react/destructuring-assignment': 0,
    'react/no-danger': 2,
    'react/prefer-stateless-function': 1,
    'react/no-array-index-key': 1,
    'react/jsx-props-no-spreading': 0,
    'react/static-property-placement': ['warn', 'property assignment', { // Указывает, где должны быть размещены статические свойства компонента React
      'childContextTypes': 'static getter',
      'contextTypes': 'static public field',
      'contextType': 'static public field',
      'displayName': 'static public field',
    }],
    'react/require-default-props': 1,
    'react/sort-comp': [1, { order: ['static-methods', 'lifecycle', 'everything-else', 'render']}],

}