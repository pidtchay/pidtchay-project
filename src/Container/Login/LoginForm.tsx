import { Form, Input, Button } from 'antd';
import * as React from 'react';
import { ILoginData } from 'Model/Authenticate';
import { ILiterals } from 'Model/Literals';
import style from 'Style/Login/LoginForm.less';
import { get } from 'Utils/common';

const layout = {
    labelCol: { span: 8 },
    wrapperCol: { span: 16 }
};

/**
 * LogIn Form props.
 */
interface ILogInProps {
    initialData: ILoginData;
    literals: ILiterals;
    onSubmit: (loginData: ILoginData) => void;
}

/**
 * LogIn Form.
 *
 * @param [ILoginData] - InitialData - LogIn Data model.
 * @param [Function] - OnSubmit - submit action.
 * @param ILoginData.initialData
 * @param ILoginData.onSubmit
 * @param ILoginData.literals
 */
export const LoginForm: React.FC<ILogInProps> = ({
    initialData,
    onSubmit,
    literals
}) => {
    const [form] = Form.useForm();

    const onFinish = (values) => {
        onSubmit({ ...values });
    };

    const onReset = () => {
        form.resetFields();
    };

    const onFill = () => {
        form.setFieldsValue(initialData);
    };

    return (
        <Form
            {...layout}
            className={style.loginForm}
            form={form}
            onFinish={onFinish}
        >
            <Form.Item
                label={get(literals, 'Forms.Login.nickname.label')}
                name="nickName"
                rules={[
                    {
                        required: true,
                        message: (
                            <>{get(literals, 'Forms.Login.nickname.message')}</>
                        )
                    }
                ]}
            >
                <Input />
            </Form.Item>
            <Form.Item
                label={get(literals, 'Forms.Login.email.label')}
                name="email"
                rules={[
                    {
                        required: true,
                        message: (
                            <>{get(literals, 'Forms.Login.email.message')}</>
                        ),
                        type: 'email'
                    }
                ]}
            >
                <Input />
            </Form.Item>
            <Form.Item
                label={get(literals, 'Forms.Login.password.label')}
                name="password"
                rules={[
                    {
                        required: true,
                        message: (
                            <>{get(literals, 'Forms.Login.password.message')}</>
                        )
                    }
                ]}
            >
                <Input.Password />
            </Form.Item>
            <Form.Item wrapperCol={{ ...layout.wrapperCol, offset: 8 }}>
                <Button type="primary" htmlType="submit">
                    {get(literals, 'ACTIONS.submit')}
                </Button>
                <Button htmlType="button" onClick={onReset}>
                    {get(literals, 'ACTIONS.reset')}
                </Button>
                <Button type="link" htmlType="button" onClick={onFill}>
                    {get(literals, 'ACTIONS.fill_form')}
                </Button>
            </Form.Item>
        </Form>
    );
};
