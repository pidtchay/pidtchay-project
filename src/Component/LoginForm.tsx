import * as React from 'react';
import { ILoginData } from 'Model/Login';
import { InputField } from './ImputField';

/**
 * LogIn Form props
 */
interface ILogInProps {
    initialData: ILoginData;
    onSubmit: (loginData: ILoginData) => void;
}

/**
 * LogIn Form
 * @param [ILoginData] initialData - LogIn Data model.
 * @param [Function] onSubmit - submit action. 
 */
export const LogInForm: React.FC<ILogInProps> = ({initialData, onSubmit}) => {
    const [loginData, setLoginData] = React.useState(initialData);
    
    const handleSubmit = (e: any) => {
        e.preventDefault();
        onSubmit(loginData);
    }

    return (
        <form onSubmit={handleSubmit}>
            <InputField label="Enter your nickname" type="text" value={loginData.nickName} onChange={e => setLoginData({...loginData, nickName: e.target.value})}/>
            <InputField label="Enter your email" type="text" value={loginData.email} onChange={e => setLoginData({...loginData, email: e.target.value})}/>
            <InputField label="Enter your password" type="text" value={loginData.password} onChange={e => setLoginData({...loginData, password: e.target.value})}/>
            <button>Submit</button>
        </form>
    );
};
