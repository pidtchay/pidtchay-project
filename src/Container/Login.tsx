import { LoginForm } from "Component/LoginForm";
import { connect } from "react-redux";
import { thunkUpdateSession } from "Store/system/thunks";
import { ILoginData } from "Model/Authenticate";
import React from "react";
import { throttle } from "../Utils/common";

const loginData: ILoginData = {
    nickName: 'Tom',
    email: 'tom@test.com',
    password: '111'
}

interface IProps {
    logInSubmit: (data: ILoginData) => void;
}

const LogInContainer = ({logInSubmit}: IProps) => {
    const handleSubmit = (data: ILoginData) => {
        throttle(() => {
            logInSubmit(data);
        }, 1000);
    }
    return(
        <div>
            <LoginForm initialData={loginData} onSubmit={handleSubmit} />
        </div>
    );
}

const mapPropsToDispatch = {
    logInSubmit: thunkUpdateSession
}

export const LogIn = connect(null, mapPropsToDispatch)(LogInContainer); 