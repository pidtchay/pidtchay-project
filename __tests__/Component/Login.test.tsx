
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ILoginData } from 'Model/Login';
import { LogInForm } from 'Component/LoginForm';

describe('LogIn form render success', () => {
    const initialData: ILoginData= {
        nickName: 'Test',
        email: 'test@test.com',
        password: '111'
    };
    const handleSubmit = (data: ILoginData) => {
        console.group('onSubmit');
        console.debug(`Your nickname: ${data.nickName}`);
        console.debug(`Your email: ${data.email}`);
        console.debug(`Your password: ${data.password}`);
        console.groupEnd();
    };
    it('should match the snapshot', () => {
        const component = renderer.create(<LogInForm initialData={initialData} onSubmit={handleSubmit} />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
});