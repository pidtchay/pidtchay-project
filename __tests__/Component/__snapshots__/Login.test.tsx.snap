// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn form render success should match the snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    Enter your nickname
    <input
      onChange={[Function]}
      type="text"
      value="Test"
    />
  </label>
  <label>
    Enter your email
    <input
      onChange={[Function]}
      type="text"
      value="test@test.com"
    />
  </label>
  <label>
    Enter your password
    <input
      onChange={[Function]}
      type="text"
      value="111"
    />
  </label>
  <button>
    Submit
  </button>
</form>
`;
